src/Emogrifier/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -58,11 +60,8 @@ private function __construct()
      *
      * @throws \InvalidArgumentException if $unprocessedHtml is anything other than a non-empty string
      */
-    public static function fromHtml($unprocessedHtml)
+    public static function fromHtml(string $unprocessedHtml): AbstractHtmlProcessor
```

Comment:
Is `: static` not possible?

Reply:
Oh, that should indeed be `static`. I think that was some fallout of the automatic code conversion.

Reply:
I've just tested it: `: static` indeed does not work. :-(

Reply:
:/ Maybe in a future PHP version,

Reply:
But would `: self` work...?

Reply:
It would. :-) #784